include/dmlc/optional.h

```diff
@@ -173,9 +176,49 @@ std::istream &operator>>(std::istream &is, optional<T> &t) {
   }
   return is;
 }
+/*! \brief specialization of '>>' istream parsing for optional<bool>
+ *
+ * Permits use of generic parameter FieldEntry<DType> class to create
+ * FieldEntry<optional<bool>> without explicit specialization.
+ *
+ *  \code
+ *    dmlc::optional<bool> x;
+ *    std::string s1 = "true";
+ *    std::istringstream is1(s1);
+ *    s1 >> x;  // x == optional<bool>(true)
+ *
+ *    std::string s2 = "None";
+ *    std::istringstream is2(s2);
+ *    s2 >> x;  // x == optional<bool>()
+ *  \endcode
+ *
+ *  \param is input stream
+ *  \param t target optional<bool> object
+ *  \return input stream
+ */
+inline std::istream &operator>>(std::istream &is, optional<bool> &t) {
+  std::string s;
+  is >> s;
```

Comment:
I can't think of a Tuple-related use that would be a problem.  Feel free to suggest one.  First of all,  optional&lt;bool&gt; are not initialized by strings.  There's a no-arg constructor and one that takes a bool reference.  The operator>> implementation is used to set an already-constructed instance.  Although this isn't exactly your concern, I've added a test to prove the following works:

    dmlc::optional&lt;bool&gt;  x, y;
    std::istringstream is("false true");
    is >> x >> y;
    CHECK_EQ(x.value(), 0);
    CHECK_EQ(y.value(), 1);

I've also verified that in the Python context, if a user does  mx.symbol.SomeOperator(...,my_opt_bool_param="false more_junk",...)

...then an error is generated.

If you're tempted to look into this further, keep in mind that the new operator>> specialization is designed to be invoked by the following code within parameter.h:
template<typename TEntry, typename DType>
class FieldEntryBase : public FieldAccessEntry {
  virtual void Set(void *head, const std::string &value) const {
    std::istringstream is(value);
    **is >> this->Get(head);**

Feel free to describe your Tuple scenario more specifically and I'll investigate further.